src/coord/hybrid/dist_utils.c

```diff
@@ -33,6 +33,23 @@ void netCursorCallback(MRIteratorCallbackCtx *ctx, MRReply *rep) {
     return;
   }
 
+  if (cmd->forProfiling && cmd->protocol == 3) {
+    MRReply *rows = NULL, *meta = NULL;
+    meta = MRReply_ArrayElement(rep, 0);
+    meta = MRReply_MapElement(meta, "results");  // profile has an extra level
```

Comment:
It would be safer to put this after the assertions block (so we are guaranteed the structure is valid)